pkg/apis/config/defaults/networking.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package defaults contains cross-api-version configuration defaults
+package defaults
+
+// APIServerAddressIPv4 is the default IPv4 address for the Networking.APIServerAddress field
+const APIServerAddressIPv4 = "127.0.0.1"
+
+// APIServerAddressIPv6 is the default IPv6 address for the Networking.APIServerAddress field
+const APIServerAddressIPv6 = "::1"
+
+// PodSubnetIPv4 is the default IPv4 subnet for the Networking.PodSubnet field
+const PodSubnetIPv4 = "10.244.0.0/16"
+
+// node-mask cidr default is /64, so we need a larger subnet, we use /56 following best practices
+// xref: https://www.ripe.net/publications/docs/ripe-690#4--size-of-end-user-prefix-assignment---48---56-or-something-else-
+
+// PodSubnetIPv6 is the default IPv6 subnet for the Networking.PodSubnet field
+const PodSubnetIPv6 = "fd00:10:244::/56"
+
+// PodSubnetDualStack is the default DualStack subnet for the Networking.ServiceSubnet field
+const PodSubnetDualStack = "10.244.0.0/16,fd00:10:244::/56"
+
+// https://github.com/kubernetes/kubernetes/blob/746404f82a28e55e0b76ffa7e40306fb88eb3317/cmd/kubeadm/app/apis/kubeadm/v1beta2/defaults.go#L32
+// Note: kubeadm is using a /12 subnet, that may allocate a 2^20 bitmap in etcd
+// we allocate a /16 subnet that allows 65535 services (current Kubernetes tested limit is O(10k) services)
+
+// ServiceSubnetIPv4 is the default IPv4 subnet for the Networking.ServiceSubnet field
+const ServiceSubnetIPv4 = "10.96.0.0/16"
+
+// ServiceSubnetIPv6 is the default IPv6 subnet for the Networking.PodSubnet field
+const ServiceSubnetIPv6 = "fd00:10:96::/112"
+
+// ServiceSubnetDualStack is the default DualStack subnet for the Networking.ServiceSubnet field
+const ServiceSubnetDualStack = "10.96.0.0/16,fd00:10:96::/112"
+
+// ExtraPortMappingListenAddressIPv4 is the default IPv4/DualStack listen address for ExtraPortMappings
+const ExtraPortMappingListenAddressIPv4 = "0.0.0.0"
```

Comment:
did we really need to make all of these public variables?

even if the values are documented, that doesn't mean we've committed to thoughtful names yet and that the specifics of how traffic works on a future provider will be the same.

Reply:
Still not thrilled about this aspect, we're exporting a lot of internal choices here that should be versioned with the API and we're doing so as a ton of unversioned globals ...

Reply:
This change does not seem necessary for the purpose of this PR?

Reply:
moved back to versioned defaulting.